src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/roundabout/ComplexRoundabout.java

```diff
@@ -0,0 +1,302 @@
+        else
+        {
+            return RoundaboutDirection.UNKNOWN;
+        }
+    }
+
+    /**
+     * This method returns the cross product between two adjacent edges.
+     *
+     * @see "https://en.wikipedia.org/wiki/Cross_product"
+     * @param edge1
+     *            An Edge entity in the roundabout
+     * @param edge2
+     *            An Edge entity in the roundabout adjacent to edge1
+     * @return A double corresponding to the cross product between two edges
+     */
+    private static double getCrossProduct(final Edge edge1, final Edge edge2)
+    {
+        // Get the nodes' latitudes and longitudes to use in deriving the vectors
+        final double node1Y = edge1.start().getLocation().getLatitude().asDegrees();
+        final double node1X = edge1.start().getLocation().getLongitude().asDegrees();
+        final double node2Y = edge1.end().getLocation().getLatitude().asDegrees();
+        final double node2X = edge1.end().getLocation().getLongitude().asDegrees();
+        final double node3Y = edge2.end().getLocation().getLatitude().asDegrees();
+        final double node3X = edge2.end().getLocation().getLongitude().asDegrees();
+
+        // Get the vectors from node 2 to 1, and node 2 to 3
+        final double vector1X = node2X - node1X;
+        final double vector1Y = node2Y - node1Y;
+        final double vector2X = node2X - node3X;
+        final double vector2Y = node2Y - node3Y;
+
+        // The cross product tells us the direction of the orthogonal vector, which is
+        // Directly related to the direction of rotation/traffic
+        return (vector1X * vector2Y) - (vector1Y * vector2X);
+    }
+
+    public ComplexRoundabout(final Edge source)
+    {
+        this(source, LEFT_DRIVING_COUNTRIES_DEFAULT);
+    }
+
+    public ComplexRoundabout(final Edge source, final List leftDrivingCountries)
+    {
+        super(source);
+        try
+        {
+            if (!(
+            // Make sure that the source has an iso_country_code
+            source.getTag(ISOCountryTag.KEY).isPresent()
+                    // Make sure that the source is an instances of a roundabout
+                    && JunctionTag.isRoundabout(source)
+                    // Make sure that we are looking at a master edge
+                    && (source.isMasterEdge())))
+            {
+                final ComplexEntityError sourceError = new ComplexEntityError(this,
+                        String.format("Invalid source Edge (%s) for a roundabout.",
+                                source.getIdentifier()),
+                        new CoreException(EXCEPTION_MESSAGE));
+                this.invalidationReasons.add(sourceError);
+                throw sourceError.getException();
+            }
+            final String isoCountryCode = source.tag(ISOCountryTag.KEY).toUpperCase();
+
+            // Get all edges in the roundabout
+            this.roundaboutEdgeSet = new SimpleEdgeWalker(source, this.isRoundaboutEdge())
```

Comment:
Has there been any use case where two roundabouts are adjacent and connected?

Reply:
The only example of that I am aware of is the Magic Roundabout in Swindon, UK. However, that is not actually tagged roundabout. In fact, the connecting of two roundabouts in that fashion would go against the practices recommended in the OSM wiki because of the multiple connections at a single node, which is discouraged for routing reasons.